app/models/sushi.rb

```diff
@@ -112,6 +112,21 @@ def rescued_last_month_available(*args)
     end
   end
 
+  ##
+  # @param params [Hash, ActionController::Parameters]
+  # @param allowed_parameters [Array<String>]
+  #
+  # @return [String]
+  # @raise [Sushi::Error::UnrecognizedParameterError] when any unrecognized params are given.
+  module ParameterValidation
+    def validate_paramaters(params = {}, allowed_parameters: [])
+      filtered_params = params.reject { |key, _| ['action', 'controller', 'format'].include?(key) }
```

Comment:
```ruby
given = params.keys.map(&:to_s) - ['action', 'controller', 'format']
return true if (given & allowed_parameters).length == given.length

 raise Sushi::Error::UnrecognizedParameterError.new(data: "The given parameter(s) are invalid: #{(given - allowed_parameters).join(', ')}.")
```

On the original line 124 you mapped the keys to_s; but if there's a mismatch, you don't remap the keys to string.  This handles that mapping ensuring you're comparing strings to strings..

Reply:
good catch. I hadn't tested for the latter statement and it didn't cross my mind.

Reply:
also, is there an actual benefit of returning true instead of just returning?

Reply:
No significant benefit…just a slight "Hey this `validate` method said true" is kind of a nice response (given that Nil is false)